### ISSUE TITLE
Implement a conversion webhook from CheCluster v1 to v2alpha1

### ISSUE THREAD

**Issue by metlos** [eclipse-che/che#19451]
This is a subtask of #19448 and depends on #19449.

The conversion webhook needs to be in place so that we can then later start storing all `CheCluster` resources in the new `v2alpha1` version. We should explore how much of the `v2alpha1` configuration can be infrerred from the `v1` configuration during the conversion from `v1` and what properties of `v2alpha1` have their equivalents in `v1`.

**Comment by metlos** [comment]
This is superseded by #19845. Once we get to the point where we need to define the new CR version, we can revisit this issue or create a new one.
